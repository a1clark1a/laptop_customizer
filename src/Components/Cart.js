import React from "react";
import { Component } from "react";
import Total from "./Total";

class Cart extends Component {
  render() {
    const { currency, selected } = this.props;
    const summary = Object.keys(selected).map((feature, idx) => {
      //Price Component
      const featureHash = feature + "-" + idx;
      const selectedOption = selected[feature];
      console.log(selectedOption);
      return (
        <div className="summary__option" key={featureHash}>
          <div className="summary__option__label">{feature} </div>
          <div className="summary__option__value">{selectedOption.name}</div>
          <div className="summary__option__cost">
            {currency.format(selectedOption.cost)}
          </div>
        </div>
      );
    });
    return (
      <section className="main__summary">
        <h2>Your cart</h2>
        {summary}
        <Total currency={currency} selected={selected} />
      </section>
    );
  }
}

Cart.defaultProps = {
  selected: {}
};

export default Cart;
