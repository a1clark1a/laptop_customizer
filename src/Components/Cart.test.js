import React from "react";
import ReactDOM from "react-dom";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Cart from "./Cart";

describe("Cart Component", () => {
  //Test state
  const state = {
    selected: {
      Processor: {
        name: "17th Generation Intel Core HB (7 Core with donut spare)",
        cost: 700
      },
      "Operating System": {
        name: "Ubuntu Linux 16.04",
        cost: 200
      },
      "Video Card": {
        name: "Toyota Corolla 1.5v",
        cost: 1150.98
      },
      Display: {
        name: '15.6" UHD (3840 x 2160) 60Hz Bright Lights and Knobs',
        cost: 1500
      }
    }
  };

  //test format
  const USCurrencyFormat = new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD"
  });

  it("renders without error", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Cart currency={USCurrencyFormat} selected={state.selected} />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it("renders given no items", () => {
    const wrapper = shallow(<Cart />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
