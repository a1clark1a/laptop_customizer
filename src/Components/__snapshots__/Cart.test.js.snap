// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cart Component renders given no items 1`] = `
<section
  className="main__summary"
>
  <h2>
    Your cart
  </h2>
  <Total
    selected={Object {}}
  />
</section>
`;
